executor/table_reader.go

```diff
@@ -244,9 +244,12 @@ func (e *TableReaderExecutor) Next(ctx context.Context, req *chunk.Chunk) error
 	// When 'select ... for update' work on a partitioned table, the table reader should
 	// add the partition ID as an extra column. The SelectLockExec need this information
 	// to construct the lock key.
-	physicalID := getPhysicalTableID(e.table)
 	if e.extraPIDColumnIndex.valid() {
-		fillExtraPIDColumn(req, e.extraPIDColumnIndex.value(), physicalID)
+		physicalID := getPhysicalTableID(e.table)
+		if physicalID != e.table.Meta().ID {
+			// table partition in static prune mode (one TableReaderExecutor per partition)
```

Comment:
In dynamic prune mode, the `getPhysicalTableID(e.table)` cannot get any other table id than the logical one, so that is why the check is here, if `physicalID` differs from `e.table.Meta().ID` then it must be a partitioned table under static pruning mode.

Reply:
So how should we handle the dynamic prune mode?
In the long run, we will make dynamic prune mode the default.

Reply:
This code has now been removed from executor/table_reader.go

In both static and dynamic partition prune mode, it will get the physical table id from store.

In the end it was too hard and high risk of inconsistencies/errors to have different handling of static and dynamic prune mode for setting the Physical Table ID (even if there is a small overhead cost that could be removed for static prune mode).